tests/src/Error.cpp

```diff
@@ -0,0 +1,234 @@
+	CHECK_IF(expect_kw) {
+		CHECK(expect_kw->kind() == ErrorKind::ExpectedKeyword);
+		CHECK(expect_kw->message() == "expected keyword"sv);
+		CHECK(expect_kw->production_name() == "production2"sv);
+		CHECK(expect_kw->annotations().empty());
+	}
+
+	auto* expect_char_c = errors.create<ExpectedCharClass>("expected char", "production3");
+	CHECK_IF(expect_char_c) {
+		CHECK(expect_char_c->kind() == ErrorKind::ExpectedCharClass);
+		CHECK(expect_char_c->message() == "expected char"sv);
+		CHECK(expect_char_c->production_name() == "production3"sv);
+		CHECK(expect_char_c->annotations().empty());
+	}
+
+	auto* generic_error = errors.create<GenericParseError>("generic error", "production 4");
+	CHECK_IF(generic_error) {
+		CHECK(generic_error->kind() == ErrorKind::GenericParseError);
+		CHECK(generic_error->message() == "generic error"sv);
+		CHECK(generic_error->production_name() == "production 4"sv);
+		CHECK(generic_error->annotations().empty());
+	}
+
+	auto* sem_error = errors.create<SemanticError>("sem error");
+	CHECK_IF(sem_error) {
+		CHECK(sem_error->kind() == ErrorKind::SemanticError);
+		CHECK(sem_error->message() == "sem error"sv);
+		CHECK(sem_error->annotations().empty());
+	}
+
+	auto* sem_warn = errors.create<SemanticWarning>("sem warn");
+	CHECK_IF(sem_warn) {
+		CHECK(sem_warn->kind() == ErrorKind::SemanticWarning);
+		CHECK(sem_warn->message() == "sem warn"sv);
+		CHECK(sem_warn->annotations().empty());
+	}
+
+	auto* sem_info = errors.create<SemanticInfo>("sem info");
+	CHECK_IF(sem_info) {
+		CHECK(sem_info->kind() == ErrorKind::SemanticInfo);
+		CHECK(sem_info->message() == "sem info"sv);
+		CHECK(sem_info->annotations().empty());
+	}
+
+	auto* sem_debug = errors.create<SemanticDebug>("sem debug");
+	CHECK_IF(sem_debug) {
+		CHECK(sem_debug->kind() == ErrorKind::SemanticDebug);
+		CHECK(sem_debug->message() == "sem debug"sv);
+		CHECK(sem_debug->annotations().empty());
+	}
+
+	auto* sem_fixit = errors.create<SemanticFixit>("sem fixit");
+	CHECK_IF(sem_fixit) {
+		CHECK(sem_fixit->kind() == ErrorKind::SemanticFixit);
+		CHECK(sem_fixit->message() == "sem fixit"sv);
+		CHECK(sem_fixit->annotations().empty());
+	}
+
+	auto* sem_help = errors.create<SemanticHelp>("sem help");
+	CHECK_IF(sem_help) {
+		CHECK(sem_help->kind() == ErrorKind::SemanticHelp);
+		CHECK(sem_help->message() == "sem help"sv);
+		CHECK(sem_help->annotations().empty());
+	}
+
+	auto* prim_annotation = errors.create<PrimaryAnnotation>("primary annotation");
+	CHECK_IF(prim_annotation) {
+		CHECK(prim_annotation->kind() == ErrorKind::PrimaryAnnotation);
+		CHECK(prim_annotation->message() == "primary annotation"sv);
+	}
+
+	auto* sec_annotation = errors.create<SecondaryAnnotation>("secondary annotation");
+	CHECK_IF(sec_annotation) {
+		CHECK(sec_annotation->kind() == ErrorKind::SecondaryAnnotation);
+		CHECK(sec_annotation->message() == "secondary annotation"sv);
+	}
+
+	if (!prim_annotation || !sec_annotation) {
```

Comment:
Could you explain this part?
return statements in tests are highly unusual.

Reply:
Eh I guess its not a big deal, in this specific case it'll repeatedly complain about the annotations being null without the return.